pkg/actions/cluster/delete.go

```diff
@@ -151,3 +153,23 @@ func checkForUndeletedStacks(stackManager manager.StackManager) error {
 
 	return nil
 }
+
+func drainAllNodegroups(cfg *api.ClusterConfig, ctl *eks.ClusterProvider, stackManager manager.StackManager, clientSet kubernetes.Interface, allStacks []manager.NodeGroupStack) error {
+	if len(allStacks) == 0 {
+		return nil
+	}
+
+	for _, s := range allStacks {
+		if s.Type == api.NodeGroupTypeUnmanaged {
+			cmdutils.PopulateUnmanagedNodegroup(s.NodeGroupName, cfg)
+		}
+	}
+
+	logger.Info("will drain %d unmanaged nodegroup(s) in cluster %q", len(cfg.NodeGroups), cfg.Metadata.Name)
+	nodeGroupManager := nodegroup.New(cfg, ctl, clientSet)
+	plan, disableEviction := false, false
+	if err := nodeGroupManager.Drain(cmdutils.ToKubeNodeGroups(cfg), plan, ctl.Provider.WaitTimeout(), disableEviction); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
missing unit tests but I recall this whole package being really hard to test :/

Reply:
ikr, this blocked me so much! I decided to let it go because it's an unexported helper func + it's wrapping functions that are all fully tested... not ideal. i could make it an exported func to test it though?

Reply:
:shrug: I think your right to just let it go, the `Drain` itself is tested which is good. I think we can flag this up in https://github.com/weaveworks/eksctl/issues/4212

Reply:
yes! good idea to link it :)